OpenRA.Game/Widgets/Widget.cs

```diff
@@ -524,8 +524,11 @@ public virtual void HideChild(Widget child)
 
 		public virtual void RemoveChildren()
 		{
-			while (Children.Count > 0)
-				RemoveChild(Children[Children.Count - 1]);
+			// PERF: Instead of removing children one by one, notify and clear in bulk
```

Comment:
Honestly that comment is pointless because clearing the collection at once should be a no-brainer. A slightly more reasonable comment would be why there is a `.Reverse()` in there. (Hint: I don't remember, but things didn't go well otherwise. If I had to guess it has to do with recalculating bounds for all elements after the removed one).
Also
```suggestion
			// PERF: Instead of removing children one by one, notify them that they are being removed and clear in bulk.
```

Reply:
Actually yeah I'd agree that it is kinda a no brainer, I'll update the PERF comment to be shorter

Reply:
This is what I meant with slowing down the PR. 

This is some ancient 11 y old code. I tried un-reversing the order and nothing out of the ordinary happened, looking through the code base, nothing requires order. The code we replaced is really just a standard design pattern. And the reason for keeping the reverse order in the PR seems:

<img width="785" alt="Screenshot 2022-08-30 at 22 41 25" src="https://user-images.githubusercontent.com/37534529/187528609-7a2d1ad5-d85a-450c-869a-9d6f0a8d2a12.png">